### PR TITLE
More Windows (MSYS) stuff

### PR DESCRIPTION
**Pull request by hgdagon**
Properties Full Info

So, the big change is the introduction of a resource file. I don't have any experience with resource files, I just followed the documentation. I was planning to draw a new logo/icon, currently, the (modified) repo banner is used as icon.

Other changes are minor. Please, review as you see fit.

------------------------------------------------------
There are 3 changes:
1. (The big one) Introduced a resource file for Windows (not integrated into VS projects).
2. Silenced the `rm` command in all makefiles (just to tidy up the output), minor cleanups related to third-party libraries.
3. Experiments with tools (static build on Windows, grammar, drag-and-drop support, fixed warning).

Not for merging, commit messages are a mess, will redo with proper messages for whatever gets accepted.

### CODE REVIEW

File: files/fheroes2.rc [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+FILEFLAGSMASK   VS_FF_DEBUG
+FILEFLAGS       FH2_DEBUG
+FILEOS          VOS_NT
+FILETYPE        VFT_APP
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040904B0"
+        BEGIN
+            VALUE "Comments",         "https://ihhub.github.io/fheroes2/\0"
```
Comment (ihhub) [inline files/fheroes2.rc:31]: Let's change it to `Visit us at https://ihhub.github.io/fheroes2`.

File: files/fheroes2.rc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            VALUE "CompanyName",      "ihhub\0" //Placeholder. What do we put here? Can be omitted.
```
Comment (ihhub) [inline files/fheroes2.rc:32]: Let's put `fheroes2 Resurrection team`.

File: files/fheroes2.rc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            VALUE "FileDescription",  "Free implementation of Heroes of Might and Magic II game engine\0"
+            VALUE "FileVersion",      FH2_VERSION_STR
+            VALUE "InternalName",     FH2_INTNAME_STR
+            VALUE "LegalCopyright",   "\251 2021 Andrey Afletdinov <fheroes2@gmail.com>\0" //Placeholder. Not sure what goes here. NWC? 3DO? Ubisoft? Can be omitted.
```
Comment (ihhub) [inline files/fheroes2.rc:36]: Please change it into `"\251 2021 fheroes2 Resurrection team <fhomm2@gmail.com>. All rights for the original HoMM II game belong to Ubisoft"`.

File: files/fheroes2.rc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            VALUE "OriginalFilename", FH2_ORGNAME_STR
+            VALUE "ProductName",      "Free Heroes2 Engine\0"
```
Comment (ihhub) [inline files/fheroes2.rc:38]: Please change it to `Free Heroes of Might and Magic II engine`.

File: files/fheroes2.rc [hunk 5]
```diff
@@ -0,0 +1,47 @@
+#include <windows.h>
```
Comment (ihhub) [inline files/fheroes2.rc:1]: Please move this whole file to `src/resources` directory.

File: src/fheroes2/system/gamedefs.h [hunk 6]
```diff
@@ -22,8 +22,10 @@
 #ifndef H2GAMEDEFS_H
 #define H2GAMEDEFS_H
 
+#ifndef RC_INVOKED
 #include "tools.h"
 #include "types.h"
+#endif
 
 #define MAJOR_VERSION 0
```
Comment (ihhub) [inline src/fheroes2/system/gamedefs.h:30]: Could we please create a new files `src/fheroes2/system/version.h` where we move all these 3 definitions? In this case we do not need the above trick and we will decouple the code.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @hgdagon , would you mind to create a new pull request with changes for `files/fheroes2.rc` and `src/fheroes2/system/gamedefs.h` files only excluding even images? These changes are useful for Visual Studio build as well and it's good to separate them from others.

File: src/fheroes2/system/gamedefs.h (hunk 6 above)
Comment (hgdagon) [reply src/fheroes2/system/gamedefs.h:30]: OK, move, as in remove them from gamedefs, right?

File: .gitignore [hunk 7]
```diff
@@ -43,9 +43,11 @@
 *.obj
 *.pch
 *.tmp
+*.res
 # visual studio directory
 .vs
 .vscode
+VisualStudio/packages/*
```
Comment (ihhub) [inline .gitignore:50]: This change won't allow us to modify `zlib1.2.11.zip` in the future.

File: script/demo/download_demo.sh [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+if ! [ '$(command -v wget)' ]; then
```
Comment (ihhub) [inline script/demo/download_demo.sh:3]: Could you please confirm that this script works on MacOS as well?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @hgdagon , I left few comments here for now. Could you please take a look?

File: script/demo/download_demo.sh (hunk 8 above)
Comment (hgdagon) [reply script/demo/download_demo.sh]: It does now!

File: .gitignore (hunk 7 above)
Comment (hgdagon) [reply .gitignore:50]: Excluded \`zlib\*.zip\`

File: src/tools/82m2wav.cpp [hunk 9]
```diff
@@ -27,18 +27,26 @@
 
 int main( int argc, char ** argv )
 {
-    if ( argc != 3 ) {
```
[the remark this answers is not in the document]
Comment (hgdagon) [reply src/tools/82m2wav.cpp:30]: Sorry, been a bit too busy recently. Seems like you made all the changes yourself)

File: script/demo/download_demo.bat [hunk 10]
```diff
@@ -0,0 +1,20 @@
+@echo off
+
+if not exist "demo" mkdir "demo"
+
+echo [1/3] downloading demo version
+powershell -Command "(New-Object Net.WebClient).DownloadFile('https://archive.org/download/HeroesofMightandMagicIITheSuccessionWars_1020/h2demo.zip', 'demo\demo.zip')"
+
+echo [2/3] unpacking archive
+powershell -command "Expand-Archive -Force 'demo\demo.zip' 'demo'"
```
[the remark this answers is not in the document]
Comment (hgdagon) [reply script/demo/download_demo.bat:9]: I think it's time for me to close this, make separate branches, and separate PRs.

File: script/demo/download_demo.bat [hunk 11]
```diff
@@ -0,0 +1,20 @@
+@echo off
```
[the remark this answers is not in the document]
Comment (hgdagon) [reply script/demo/download_demo.bat:1]: No, just for consistency - ``download_demo.sh'/`download_demo.bat``. Why, do we need to specify the os?

**Comment by ihhub** [comment]
Hi @hgdagon ,
let's do these steps:
1. Keep this pull request open but change it to draft
2. Open a new pull request just for package script installation (with changes for appveyor)
3. Open a new pull request for demo script changes (this is the most important). Please try to use `git mv` command for file renaming before modifying them. It will help to prevent growing git size repository
4. Open a new pull request just for tools fixes
5. Open a new pull request for MinGW changes.


What do you think?

File: script/windows/install_packages.bat [hunk 12]
```diff
@@ -1,76 +1,55 @@
 @echo off
 
-if not exist "..\..\..\packages"           mkdir "..\..\..\packages"
```
Comment (ihhub) [inline script/windows/install_packages.bat:3]: The problem with using this approach is that user with Win 7 or older won't be able to run it if PowerShell of version lower than 3 is installed.

**Comment by ihhub** [comment]
@hgdagon , I'm closing this pull request as most of the changes we did in a separate pull request. Please feel free to create a new pull request if we miss something.

**Pull request closed without merging**